### DISCUSSION TITLE
提示词翻译器最佳实践 - Best practices for prompt word translators

### DISCUSSION THREAD

**Discussion by metercai** [lllyasviel/Fooocus#1660] [Show and tell]
提示词翻译器是萌发SimpleSDXL项目的最初诱因。经过前段时间的实践总结，将提示词翻译器进行了升级，应该是目前的最佳实践了：
- 支持中英文混合编辑，对文本进行中英文切分后分别翻译再合并，适配提示词这种标签为主的表达场景。
- 支持翻译后再编辑。不管哪种翻译方式，结果质量都是不可控的，存在翻译质量差导致生成内容偏差的现象。翻译后再编辑可以显性化问题，提供用户再调整处理的空间。
- 引入第三方在线翻译，由用户自主选择本地模型翻译还是第三方接口翻译。毕竟本地模型翻译的代价偏高，用户的使用环境千差万别，可用性是第一位的。
- 选择经过验证的大厂稳定接口，每次启动时随机选择，运行态相对固定。既避免对接口冲击又保持翻译的一致性。

新版本升级已上线，可以实际体验体验。https://github.com/metercai/SimpleSDXL
<img width="414" alt="Best practices for prompt word translators" src="https://github.com/lllyasviel/Fooocus/assets/5652458/9aade547-6b58-4797-b845-776722fd9494">

**Comment by mashb1t** [comment]
Great work! Relates to https://github.com/lllyasviel/Fooocus/pull/1039 but local translation is always better concerning data privacy!

**Reply by metercai** [reply]
还在继续实践中，争取把本地翻译的模型尺寸降下来，提高低配用户的可用性。

**Comment by Irmagon** [comment]
A friend of mine wrote a little translation extension.

https://github.com/AlekPet/Fooocus_Extensions_AlekPet

**Reply by mashb1t** [reply]
basically that's what https://github.com/lllyasviel/Fooocus/pull/1039 does, gj!

**Reply by metercai** [reply]
It relies on GoogleTranslator and is not suitable for Chinese mainland

**Reply by mashb1t** [reply]
would you mind contributing your implementation to Fooocus by opening a PR?

**Reply by metercai** [reply]
No mind, but it's challenging to unbinding and contributing to Fooocus from SimpleSDXL:
<img width="384" src="https://github.com/lllyasviel/Fooocus/assets/5652458/a67026b5-8d79-4e57-ab6b-e7ce44e55a8b">
Give me some time to try it out.
